governance/program/src/instruction.rs

```diff
@@ -0,0 +1,264 @@
+    ///   0. `[writable]` Governance voting record account
+    ///                   Can be uninitialized or initialized(if already used once in this proposal)
+    ///                   Must have address with PDA having seed tuple [Governance acct key, proposal key, your voting account key]
+    ///   1. `[writable]` Proposal State account
+    ///   2. `[writable]` Your Vote account.
+    ///   3. `[writable]` Your Yes Vote account
+    ///   4. `[writable]` Your No Vote account
+    ///   5. `[writable]` Vote Mint account
+    ///   6. `[writable]` Yes Vote Mint account
+    ///   7. `[writable]` No Vote Mint account
+    ///   8. `[]` Source Token Mint account
+    ///   9. `[]` Proposal account
+    ///   10. `[]` Governance account
+    ///   12. `[]` Transfer authority
+    ///   13. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   14. `[]` Token program account
+    ///   15. `[]` Clock sysvar
+    Vote {
+        /// Yes/No  with amount of votes
+        vote: Vote,
+    },
+
+    /// Executes a command in the Proposal
+    ///
+    ///   0. `[writable]` Transaction account you wish to execute
+    ///   1. `[writable]` Proposal State account
+    ///   2. `[]` Program being invoked account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Governance account
+    ///   5. `[]` Governance program account pub key
+    ///   6. `[]` Clock sysvar
+    ///   7+ Any extra accounts that are part of the instruction, in order
+    Execute,
+
+    /// Creates an empty governance vote record
+    ///
+    ///   0. `[]` Governance Vote Record account. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, proposal key, your voting account key
+    ///   1. `[]` Proposal key
+    ///   2. `[]` Your Vote account
+    ///   3. `[]` Payer
+    ///   4. `[]` System account.
+    CreateEmptyGovernanceVoteRecord,
+
+    /// [Requires tokens of the Governance Mint or Council Mint depending on type of Proposal]
+    /// Deposits vote tokens to be used during the voting process on a Proposal.
+    /// These tokens are removed from your account and can be returned by withdrawing
+    /// them from the Proposal (but then you will miss the vote.)
+    ///
+    ///   0. `[writable]` Governance Vote Record account. See Vote docs for more detail
+    ///   1. `[writable]` Initialized Vote account to hold your received voting tokens
+    ///   2. `[writable]` User Source Token account to deposit tokens from
+    ///   3. `[writable]` Source Token Holding account for Proposal that will accept the tokens in escrow
+    ///   4. `[writable]` Vote Mint account
+    ///   5. `[]` Proposal account.
+    ///   6. `[]` Transfer authority
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   8. `[]` Token program account
+    DepositSourceTokens {
+        /// How many voting tokens to deposit
+        voting_token_amount: u64,
+    },
```

Comment:
Seems simpler to not have this :smile:

Reply:
Correct. It was leftover from the old workflow when source tokens where explicitly deposited. I removed it.